fluid/neural_machine_translation/transformer/train.py

```diff
@@ -260,9 +260,7 @@ def split_data(data, num_part):
 def test_context(train_progm, avg_cost, train_exe, dev_count, data_input_names,
                  sum_cost, token_num):
     # Context to do validation.
-    test_program = train_progm.clone()
-    with fluid.program_guard(test_program):
-        test_program = fluid.io.get_inference_program([avg_cost])
+    test_program = train_progm.clone(for_test=True)
```

Comment:
this might not work since this seems to be full train program

Reply:
Done. put `clone(for_test=True)` before optimization phase. Similar with https://github.com/PaddlePaddle/Paddle/pull/13498/commits/44ba6942d12662c4e5b4a425142604b28d99de88